R/utils.R

```diff
@@ -548,12 +548,19 @@ download_with_retry <- function(...) {
 #'
 #' Returns `TRUE` if the model appears to be finished running and `FALSE` otherwise.
 #' @param .mod A `bbi_nonmem_model` object.
+#' @param ... Arguments passed to methods.
 #'
+#' @export
+check_nonmem_finished <- function(.mod, ...) {
+  UseMethod("check_nonmem_finished")
+}
+
+#' @rdname check_nonmem_finished
 #' @importFrom readr read_lines
 #' @importFrom stringr str_detect
 #'
 #' @export
-check_nonmem_finished <- function(.mod) {
+check_nonmem_finished.bbi_nonmem_model <- function(.mod, ...) {
```

Comment:
I was thinking about what happens if a user _does_ pass a path here, and how that would change now that it's a method. Turns out we just get a slightly friendlier error now.

<details><summary>before these changes</summary>

```
> get_output_dir("naw")
Error in UseMethod("get_output_dir") : 
  no applicable method for 'get_output_dir' applied to an object of class "character"
> check_nonmem_finished("naw")
Error in UseMethod("get_output_dir") :
no applicable method for 'get_output_dir' applied to an object of class "character"
```

</details>

<details><summary>after these changes</summary>

```
> get_output_dir("naw")
Error in UseMethod("get_output_dir") : 
  no applicable method for 'get_output_dir' applied to an object of class "character"
> check_nonmem_finished("naw")
Error in UseMethod("check_nonmem_finished") : 
  no applicable method for 'check_nonmem_finished' applied to an object of class "character"
```
</details>

Reply:
Thanks for documenting.